### ISSUE TITLE
Lost max_xxx_data frame

### ISSUE THREAD

**Issue by scw00** [ngtcp2/ngtcp2#73]
Hi 

Tested ngtcp2 in 20 % loss env.

There are some issues
 - Rtt we measured is too large (force to send ack frame for two full-size packet,  as developed spec, can fix that).
 - max_xxx_data may lost that will cause peer blocked and hardly to recover since the subsequent packets are ack-only packets(I try to combine ack frame and max_xxx_data frame into one pkt and it work well).

Leave this issue here, and will push some psr on this weekend.

**Comment by tatsuhiro-t** [comment]
max_stream_data and max_data must be retransmitted, therefore there should be no deadlock for STREAM frame.

**Comment by scw00** [comment]
> there should be no deadlock for STREAM frame.

I don't understand, why can it be deadlock for STREAM frame.

**Comment by scw00** [comment]
Yes it should be retransmitted, but max_stream_data and max_data are retransmitted after rto timer fired. That greatly impact the sender's performance.

**Comment by tatsuhiro-t** [comment]
I haven't checked there is a bug in TLP/RTO, but 20% packet loss itself reduces throughput.
The primary reason of TLP/RTO is get acknowledgement from peer.  If it does not work and packet is always retransmitted after RTO, then I suspect that there is a bug in retransmission code.

**Comment by tatsuhiro-t** [comment]
The out of order ack only packet is processed as usual.  We don't need special treatment for it.
The spec is right.  If we generate ACK for ACK only packet, then we get infinite ACK - ACK loop which makes internet explode.

**Comment by scw00** [comment]
So we need to elicit the peer's ack by adding some other frame into packet. Otherwise rto is too expensive.
